src/client.rs

```diff
@@ -22,11 +22,87 @@ struct Host {
     token: String,
 }
 
-/// Reads the contents of a hosts.toml file located in `dir`.
-fn read_hosts_toml(mut dir: PathBuf) -> Result<Hosts> {
-    dir.push("hosts.toml");
-    let hosts = std::fs::read_to_string(dir)?;
-    Ok(toml::from_str(&hosts)?)
+/// The contents of an Oxide CLI `credentials.toml` file.
+#[derive(Debug, Deserialize, Serialize)]
+struct Credentials {
+    /// A map from host names to per-host token and user information.
+    profile: HashMap<String, Credential>,
+}
+
+/// The contents of an Oxide CLI `credentials.toml` file.
+#[derive(Debug, Deserialize, Serialize)]
+struct Credential {
+    /// The ID of the user session for this entry.
+    user: String,
+
+    /// The URL of the host for this entry.
+    host: String,
+
+    /// The authentication token associated with this entry's session.
+    token: String,
+}
+
+/// An abstraction over `credentials.toml` and `hosts.toml` files.
+struct LoginConfig {
+    dir: PathBuf,
+}
+
+impl LoginConfig {
+    /// Checks whether `credentials.toml` or `hosts.toml` exist.
+    pub fn exists(&self) -> bool {
+        let creds = self.dir.clone().join("credentials.toml");
+        let hosts = self.dir.clone().join("hosts.toml");
+
+        creds.exists() || hosts.exists()
+    }
+
+    /// Read `credentials.toml` in, falling back to `hosts.toml if not present.
+    pub fn read_config(&self) -> Result<Hosts> {
+        let hosts = self.read_credentials_toml()?;
+        if let Some(hosts) = hosts {
+            return Ok(hosts);
+        }
+        self.read_hosts_toml()
+    }
+
+    /// The name of the config file being used for credentials.
+    pub fn file_name(&self) -> &str {
+        let creds = self.dir.clone().join("credentials.toml");
+        let hosts = self.dir.clone().join("hosts.toml");
+        match (creds.exists(), hosts.exists()) {
+            (true, _) => "credentials.toml",
+            (_, true) => "hosts.toml",
+            (false, false) => "no config found",
```

Comment:
This kind of makes me feel like the function should probably return an `Option` (or  `Result`?) rather than a string for "no config found"?

Reply:
I think this was a sign that we needed tighter logic around the type of config found. I've added an enum to capture this.